pkg/controller/nodeconfig/sync_daemonsets.go

```diff
@@ -19,11 +21,14 @@ func (ncc *Controller) syncDaemonSet(
 	soc *scyllav1alpha1.ScyllaOperatorConfig,
 	daemonSets map[string]*appsv1.DaemonSet,
 ) error {
-	scyllaUtilsImage := soc.Spec.ScyllaUtilsImage
-	// FIXME: check that its not empty, emit event
-	// FIXME: add webhook validation for the format
+	if soc.Status.ScyllaDBUtilsImage == nil || len(*soc.Status.ScyllaDBUtilsImage) == 0 {
+		ncc.eventRecorder.Event(nc, corev1.EventTypeNormal, "MissingScyllaUtilsImage", "ScyllaOperatorConfig doesn't yet have scyllaUtilsImage available in the status.")
+		return controllertools.NewNonRetriable("scylla operator config doesn't yet have scyllaUtilsImage available in the status")
+	}
+	scyllaDBUtilsImage := *soc.Status.ScyllaDBUtilsImage
```

Comment:
shouldn't this validate if generation == observedGeneration first?

Reply:
Not really, it is a different resource and not dependent on the input + it reacts to updates. (It also avoid the possibly infinite loop by waiting for something to catch up. It's better to deploy what we have and  reconcile the updates.)